src/test/java/io/stargate/sgv2/jsonapi/service/projection/DocumentProjectorTest.java

```diff
@@ -132,4 +132,266 @@ public void verifyProjectionEquality() throws Exception {
+                  """
+                      { "values" : [ {
+                           "y": 2
+                        }, {
+                           "y": false,
+                           "z": true
+                        } ],
+                        "array3": [2, 3]
+                      }
+                      """));
+    }
+  }
+
+  @Nested
+  class ProjectorApplyExclusions {
+    @Test
+    public void testExcludeWithIdIncluded() throws Exception {
+      final JsonNode doc =
+          objectMapper.readTree(
+              """
+                      {  "_id" : 123,
+                         "value1" : true,
+                         "value2" : false,
+                         "nested" : {
+                            "x": 3,
+                            "y": 4,
+                            "z": -1
+                         },
+                         "nested2" : {
+                            "z": 5
+                         }
+                      }
+                      """);
+      DocumentProjector projection =
+          DocumentProjector.createFromDefinition(
+              objectMapper.readTree(
+                  """
+                                  {
+                                    "value1" : 0,
+                                    "nested" : {
+                                       "x": 0
+                                    },
+                                    "nested.z": 0,
+                                    "nosuchprop": 0
+                                  }
+                                  """));
+      assertThat(projection.isInclusion()).isFalse();
+      projection.applyProjection(doc);
+      assertThat(doc)
+          .isEqualTo(
+              objectMapper.readTree(
+                  """
+                          {
+                             "_id" : 123,
+                             "value2" : false,
+                             "nested" : {
+                                "y": 4
+                             },
+                             "nested2" : {
+                               "z": 5
+                             }
+                          }
+                                  """));
+    }
+
+    @Test
+    public void testExcludeWithIdExcluded() throws Exception {
+      final JsonNode doc =
+          objectMapper.readTree(
+              """
+                      { "_id" : 123,
+                         "value1" : true,
+                         "nested" : {
+                            "x": 3,
+                            "z": -1
+                         },
+                         "nested2" : {
+                            "z": 5
+                         }
+                      }
+                      """);
+      DocumentProjector projection =
+          DocumentProjector.createFromDefinition(
+              objectMapper.readTree(
+                  """
+                                  {
+                                    "_id": 0,
```

Comment:
@vkarpov15 and exclusion of "_id" tested also here (one for otherwise including, others for excluding)